frontend/src/views/camp/Dashboard.vue

```diff
@@ -347,25 +356,44 @@ export default {
     multiplePeriods() {
       return Object.keys(this.periods).length > 1
     },
+    ...mapGetters({
+      loggedInUser: 'getLoggedInUser',
+    }),
+  },
+  watch: {
+    'filter.category': 'persistRouterState',
+    'filter.responsible': 'persistRouterState',
+    'filter.period': 'persistRouterState',
+    'filter.progressLabel': 'persistRouterState',
   },
   async mounted() {
     this.api.reload(this.camp())
 
-    const [loggedInUser] = await Promise.all([
-      this.$auth.loadUser(),
-      this.camp().periods()._meta.load,
-      this.camp().activities()._meta.load,
-      this.camp().categories()._meta.load,
-      this.camp().progressLabels()._meta.load,
-    ])
+    const { categories, periods, collaborators, progressLabels } =
+      await loadAndProcessCollections(this.camp())
+    const queryFilters = processRouteQuery(this.$route.query)
+    const { period, responsible, category, progressLabel } = {
+      ...this.filter,
+      ...queryFilters,
+    }
+    this.filter = {
+      category: category.filter((value) => categories.includes(value)),
+      responsible: responsible.filter((value) => collaborators.includes(value)),
+      progressLabel: progressLabel.filter((value) => progressLabels.includes(value)),
+      period: periods.includes(period) ? period : null,
+    }
 
-    this.loggedInUser = loggedInUser
     this.loading = false
   },
   methods: {
     campCollaborationDisplayName(campCollaboration) {
       return campCollaborationDisplayName(campCollaboration, this.$tc.bind(this))
     },
+    persistRouterState() {
+      const query = transformValuesToHalId(this.filter)
+      if (filterAndQueryAreEqual(query, this.$route.query)) return
+      this.$router.replace({ query }).catch((err) => console.warn(err))
```

Comment:
the fix is on line 394, I implemented an equals check that doesn't trigger as easily. The issue is that the filter contains values such as `null` or `[]` which should be equal to `undefined` in the `route.query` but the check before did not take that into account. Also the `route.query` doesn't provide an array if there is only one value. The `window.history` fix just doesn't refresh navigation. But now it may refresh navigation because it's only called when there is a real value change.

Reply:
Ah I see. Thanks for clarifying